sparse/model.py

```
@@ -0,0 +1,1419 @@
+        Compute sliding window attention.
+
+        Args:
+            q: Query (B, H, N, D)
+            k: Key (B, H, N, D)
+            v: Value (B, H, N, D)
+            mask: Optional attention mask
+
+        Returns:
+            Output tensor (B, H, N, D)
+        """
+        B, H, N, D = q.shape
```

Ruff (0.14.8)

Unpacked variable `B` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)